### ISSUE TITLE
REQUEST - Incoming Comms indication

### ISSUE THREAD

**Issue by Wrongtown** [daid/EmptyEpsilon#94]
Had a session with a crew on the weekend where one player doubled as Relay + Science (using both screens, not using Operations) and they missed a lot of messages while trying to run Science scans.

Possibly improvements:

a) have Relay "tab" flashes if that screen does not have focus and there's an incoming hail waiting to be addressed.
b) have some kind of indicator on main screen.

**Comment by daid** [comment]
I think I like solution 2.

For example, in the corner a box with "INCOMMING COMMS" and "COMMS ACTIVE" for example.

**Comment by Wrongtown** [comment]
I agree, maybe it does steal a bit of the thunder from Relay announcing it, but it could also remind the crew "Who is on Relay? No-one? Right, better fix that I guess..."

**Comment by kwadroke** [comment]
If a notification like solution is created, it could be nice if that was in options.ini to enable/disable notification on the main screen.

**Comment by Wrongtown** [comment]
:+1:

**Comment by daid** [comment]
For now, I think I do nothing. As you should have ran operations instead of science+comms ;-)

I'll think about it for a bit. In the back of my head there is still the "On screen!" command that we lack, where the captain can opt to show communications on the main screen. But that also requires a better communication log.

**Comment by nallath** [comment]
A better com log would be good I think, as we could also add some generic events in there (Enemies destroyed, friendlies destroyed, etc). This way you can still miss a friendly station being destroyed, but your relay officer can figure out what happend afterwards.

**Comment by zorilya** [comment]
What about a sound... On a bridge the incoming message would surely make a sound somewhere... I like the idea of having options but i can see a sound indicating a ship is being hailed and perhaps the relevant tab flashing would be a good addition. Thought train for new players might be "Why is that sound happening, and why is that tab fla.... OH!! (comm message work flow learned)".

**Comment by zorilya** [comment]
On the subject of comms. Is there a feeling on having general comms chatter... with the flexibility this project has over something like artemis it would be possible to generate messages from and to ships / locations and the messages to be displayed in something akin to a chat log. 

From a story point of view neutral ships could be firstly more interesting, giving character and the illusion of individual an agenda whilst also allowing GMs to drop hints about the area. Say the mission is initially to investigate the sector for the presense of an emerging phenomenon that has been picked up on LRS and on their travels a mining ship exits a nebula reporting to a fleet mate that something messed up was happening to their sensors, more than usual, That is a more discreet and immersive way of dropping details rather than the block message from a station "We have detected strange readings in sector blah blah, please investigate".

**Comment by daid** [comment]
https://i.imgur.com/Djum6Cr.png
5
Making some progress.

**Comment by Wrongtown** [comment]
This is really an excellent approach, no more closing the comms message and then having the crew say "Hang on what sector did they say?"

I don't expect every scenario to utilize it fully but the idea of basic radio chatter contributing to plot is a good one.

Suggestion for an add-on down the track would be the possibility of encountering encrypted (unintelligible) comms from enemy ships. Comms can advise Science, who enter the ship call sign reference and play their scanning minigame to decrypt it. On successful decryption the comms log could show the messages from that ship properly. Of course I'm not proposing ACTUAL encryption, just utter random nonsense while the shipCommsEncryption flag has been set to 1 for a SpaceObject.

I'd suggest entering the callsign from Comms log as it makes sense to me that often this would be intercepted comms of ships which science can't see yet, so can't select for normal scans.

Apologies if that's unclear, wrote it on my phone :)

**Comment by zorilya** [comment]
Looks great! Perhaps that one for the ships log and another on the relay station for general comms chatter between ships to help create a more alive feel to the universe.
